### PR TITLE
[EFR32] Rework NVM3 instances and refactor our KVS implementation to use nvm3

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
EFR32 matter examples create a second instance of nvm3 which causes a significant overhead and flash usage impact.
The KVS platform implementation also creates its own flash section using a few flash pages independently from the already 2 existing nvm sections (nvm3 default + chip_nvm3).

#### Change overview
- Delete the chip_nvm3 section and instead use the already existing Silabs default nvm3 section. (share space with other stacks).
- Rework the section placement in the linker files.
- Create a new key domain in this nvm3 section for matter storage. 
- Refactor KVS platform implementation to use Silabs nvm3 storage.
 

#### Testing
Run persistent-storage app on EFR32 for unit testing of the new KVS platform API
Build/Run/Commission/reboot/send commands on all the example apps supported by the EFR32 platform

### CODE REVIEW

File: src/platform/EFR32/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -249,6 +257,8 @@ void BLEManagerImpl::bluetoothStackEventHandler(void * p_arg)
                 RAIL_GetVersion(&railVer, true);
                 ChipLogProgress(DeviceLayer, "RAIL version:, v%d.%d.%d-b%d", railVer.major, railVer.minor, railVer.rev,
                                 railVer.build);
+                sl_bt_connection_set_default_parameters(BLE_CONFIG_MIN_INTERVAL, BLE_CONFIG_MAX_INTERVAL, BLE_CONFIG_LATENCY,
```
Comment (selissia) [inline src/platform/EFR32/BLEManagerImpl.cpp:260]: Is this related to NVM/KVS somehow?

Comment (jmartinez-silabs) [reply src/platform/EFR32/BLEManagerImpl.cpp:260]: Yes, I have seen some BLE supervisor timeouts with the refactor. This allows a longer timeout/interval for ble communication. I did improve the storage API after that so it is probably not needed anymore but this doesn't hurt either way

**Review by selissia — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Are the codesize measurements just not catching the sections that get smaller?  They seem to mostly show .bss and .text increases....

**Comment by jmartinez-silabs** [comment]
> Are the codesize measurements just not catching the sections that get smaller?  They seem to mostly show .bss and .text increases....

@bzbarsky-apple yes it's a separate section/symbol in the linker before named nvm3_dummy regrouping the 2 nvm3 instances. Now it is just one instance called .nvm saves around 32k of flash + kvs what just allocating another 4 pages that weren't factored in .txt either
